File: src/main/java/com/sica/backend/entity/Student.java
package com.sica.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "students")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Student {

    @Id
    private String personId;

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "person_id")
    private Person person;

    @Column(nullable = false, unique = true)
    private String enrollmentNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Level level;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "division_id", nullable = false)
    private Division division;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "career_id", nullable = false)
    private Career career;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Shift shift;
}

File: src/main/kotlin/com/sica/backend/entity/Student.kt
package com.sica.backend.entity

import jakarta.persistence.*

@Entity
@Table(name = "students")
data class Student(
    @Id
    val personId: String,

    @OneToOne(fetch = FetchType.LAZY)
    @MapsId
    @JoinColumn(name = "person_id")
    val person: Person,

    @Column(nullable = false, unique = true)
    val enrollmentNumber: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val level: Level,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "division_id", nullable = false)
    val division: Division,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "career_id", nullable = false)
    val career: Career,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val shift: Shift
)
